bc_obps/service/report_service.py

```diff
@@ -1,19 +1,49 @@
 from uuid import UUID
+from django.db.models import Case, When, Value, BooleanField
 from django.db import transaction
 from django.db.models import QuerySet
-from django.db.models import Case, When, Value, BooleanField
-from registration.models import Activity, RegulatedProduct
+from registration.models import Activity
+from registration.models import RegulatedProduct
 from registration.models.operation import Operation
 from reporting.models import ReportOperationRepresentative
 from reporting.models.report import Report
 from reporting.models.facility_report import FacilityReport
 from reporting.models.report_operation import ReportOperation
 from reporting.models.report_version import ReportVersion
-from reporting.schema.report_operation import ReportOperationIn
 from service.data_access_service.report_service import ReportDataAccessService
 from service.data_access_service.reporting_year import ReportingYearDataAccessService
 from service.report_version_service import ReportVersionService
 from django.forms.models import model_to_dict
+from service.facility_report_service import FacilityReportService, SaveFacilityReportData
+from typing import Any, List, Optional
+
+
+class SaveReportOperationData:
+    def __init__(
+        self,
+        operator_legal_name: str,
+        operation_name: str,
+        operation_type: str,
+        registration_purpose: str,
+        bc_obps_regulated_operation_id: str,
+        activities: List[int],
+        regulated_products: List[int],
+        operation_report_type: str,
+        operation_representative_name: List[int],
+        operator_trade_name: Optional[str] = None,
+        operation_bcghgid: Optional[str] = None,
+    ):
+        self.operator_legal_name = operator_legal_name
+        self.operation_name = operation_name
+        self.operation_type = operation_type
+        self.registration_purpose = registration_purpose
+        self.bc_obps_regulated_operation_id = bc_obps_regulated_operation_id
+        self.activities = activities
+        self.regulated_products = regulated_products
+        self.operation_report_type = operation_report_type
+        self.operation_representative_name = operation_representative_name
+        self.operator_trade_name = operator_trade_name
+        self.operation_bcghgid = operation_bcghgid
```

Comment:
Dataclasses are friends now! They build constructors, equality, and a bunch of other things like frozen/read-only, ... automatically, we just need to list which fields.

https://docs.python.org/3/library/dataclasses.html

```suggestion
@dataclass
class SaveReportOperationData:
        operator_legal_name: str,
        operation_name: str,
        operation_type: str,
        registration_purpose: str,
        bc_obps_regulated_operation_id: str,
        activities: List[int],
        regulated_products: List[int],
        operation_report_type: str,
        operation_representative_name: List[int],
        operator_trade_name: Optional[str] = None,
        operation_bcghgid: Optional[str] = None,
    
```

Reply:
Nice! I'll update this in the follow up PR for https://github.com/bcgov/cas-reporting/issues/650